tensorflow/core/common_runtime/gpu/gpu_device_test.cc

```diff
@@ -114,6 +117,33 @@ class GPUDeviceTest : public ::testing::Test {
   }
 };
 
+TEST_F(GPUDeviceTest, CudaMallocAsync) {
+  SessionOptions opts = MakeSessionOptions("0", 0, 1, {}, {},
+                                           /*use_cuda_malloc_async=*/true);
+  std::vector<std::unique_ptr<Device>> devices;
+  Status status;
+  int number_instantiated = GpuCudaMallocAsyncAllocator::GetInstantiatedCountTestOnly();
+  { // The new scope is to trigger the destruction of the object.
+    status = DeviceFactory::GetFactory("GPU")->CreateDevices(
+        opts, kDeviceNamePrefix, &devices);
+    EXPECT_EQ(devices.size(), 1);
+    Device* device = devices[0].get();
+    auto* device_info = device->tensorflow_gpu_device_info();
+    CHECK(device_info);
```

Comment:
Could you please use `DCHECK` instead? We don't use `CHECK` in new code anymore. Sorry we didn't notice this earlier!

Reply:
This is a test. I really want a crash even in no-debug mode.
So I pushed a change to use an EXPECT instead.